.github/workflows/dockerbuild.yml

```diff
@@ -1,16 +1,50 @@
-name: Trigger Docker Build
+name: Docker Build
 
 on:
   push:
-    branches: 
-      - 'master'
+    branches:
+      - master
+  pull_request:
+  release:
+    types:
+      - created
+
 jobs:
   build:
     runs-on: ubuntu-latest
     steps:
-       - name: Repository Dispatch
-         uses: peter-evans/repository-dispatch@v1
-         with:
-           token: ${{ secrets.DOCKER_REPO_TOKEN }}
-           repository: robertsLando/Zwave2Mqtt-docker
-           event-type: build
+      - uses: actions/checkout@v2
+
+      - name: Just build on PR
+        if: ${{ github.event_name == 'pull_request' }}
+        run: docker build .
+
+      - name: Login to dockerhub
+        if: ${{ github.event_name != 'pull_request' }}
+        run: |
+          docker login --username $DOCKER_USERNAME --password $DOCKER_PASSWORD
+        env:
+          DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
+          DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
+
+      - uses: crazy-max/ghaction-docker-buildx@v1
+        if: ${{ github.event_name != 'pull_request' }}
+
+      - name: Dev
+        if: ${{ github.ref == 'git/refs/heads/master'}}
+        run: |
+          docker buildx build \
+          --platform linux/arm64/v8,linux/amd64,linux/arm/v6,linux/arm/v7,linux/386 \
+          -t ${GITHUB_REPOSITORY}:latest \
+          --push \
+          .
+
+      - name: Latest & tag
+        if: ${{ github.event_name == 'release' }}
+        run: |
+          docker buildx build \
```

Comment:
Do you know if, in case of a release, this command will build again all images or will be able to get them from cache from previous buildx command?

Reply:
it won't cache :(

Reply:
So I suggest to use: 

```bash
TAGS="-t ${GITHUB_REPOSITORY}:dev"


if [ "$GITHUB_EVENT_NAME" == "release" ]; then
 VERSION=$(echo ${GITHUB_REF} | sed "s/git\/refs\/tags\/v//")
 TAGS="$TAGS -t ${GITHUB_REPOSITORY}:latest -t ${GITHUB_REPOSITORY}:${VERSION}" # make a new release
fi

docker buildx build \
         --platform linux/arm64/v8,linux/amd64,linux/arm/v6,linux/arm/v7,linux/386 \
         $TAGS \
```

Reply:
the issue is that it'll be triggered from different events, if you're quick _(~1hr based on current build time)_ at doing a release after a master merge maybe even concurrent, so a shell script with more logic wouldn't help here.
i guess if you waited for `master/dev` to build, the `latest`+`release` could be retagging the `:dev` tag.
so a cross arch pull, tag, push
buildx has awful docs at best, lacking features at worst :(

Reply:
Based on this, what about create 2 different CI so? One that triggers on release (latest and version) and one on every master push (dev)? 

As you said, if for example we push a release to master followed by another push before the first one ends the second could prevent that the docker release its made?

Reply:
that is basically what happens, theres some `if` statements in there that play the right steps

it's not going to prevent anything, just be perceivably out of order if say the release finishes before the master branch does